app/javascript/mastodon/models/notification_group.ts

```diff
@@ -171,31 +176,33 @@ export function createNotificationGroupFromJSON(
       const { annual_report, ...groupWithoutAnnualReport } = group;
       return {
         ...groupWithoutAnnualReport,
+        partial: false,
         annualReport: createAnnualReportEventFromJSON(annual_report),
         sampleAccountIds,
       };
     }
     default:
       return {
         sampleAccountIds,
+        partial: false,
         ...group,
       };
   }
 }
 
 export function createNotificationGroupFromNotificationJSON(
   notification: ApiNotificationJSON,
-) {
+): NotificationGroup {
   const group = {
     sampleAccountIds: [notification.account.id],
     group_key: notification.group_key,
     notifications_count: 1,
-    type: notification.type,
     most_recent_notification_id: notification.id,
     page_min_id: notification.id,
     page_max_id: notification.id,
     latest_page_notification_at: notification.created_at,
-  } as NotificationGroup;
+    partial: true,
```

Comment:
Should this be hardcoded here? I have a feeling that this function could be use in both cases, and the caller knows if its partial or not, so I would add it as a function parameter?